### PR TITLE
Moving smoothing into the audio level module.

### PR DESCRIPTION
**Pull request by boks1971**

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -975,11 +974,13 @@ func (p *ParticipantImpl) onMediaTrack(track *webrtc.TrackRemote, rtpReceiver *w
...
-		"trackID", publishedTrack.ID(),
-		"rid", track.RID(),
-		"SSRC", track.SSRC())
+	if publishedTrack != nil {
+		p.params.Logger.Infow("mediaTrack published",
+			"kind", track.Kind().String(),
+			"trackID", publishedTrack.ID(),
+			"rid", track.RID(),
+			"SSRC", track.SSRC())
+	}
```
Comment (boks1971) [inline pkg/rtc/participant.go:983]: This is the crash @lukasIO reported with publish failing due to no pending track

File: pkg/sfu/audio/audiolevel.go [hunk 2]
```diff
@@ -48,34 +58,44 @@ func (l *AudioLevel) Observe(level uint8, durationMs uint32) {
...
+			// > 0 if active for longer than observe duration
+			// < 0 if active for less than observe duration
+			activityWeight := 20 * math.Log10(float64(l.activeDuration)/float64(l.params.ObserveDuration))
+			adjustedLevel := float64(l.loudestObservedLevel) + activityWeight
+			linearLevel := ConvertAudioLevel(adjustedLevel)
+
+			// exponential smoothing to dampen transients
+			smoothedLevel := l.smoothedLevel.Load()
+			smoothedLevel += (linearLevel - smoothedLevel) * l.smoothFactor
+			smoothedLevel = math.Ceil(smoothedLevel*audioLevelQuantization) * invAudioLevelQuantization
```
Comment (boks1971) [inline pkg/sfu/audio/audiolevel.go:80]: Have no idea what this quantisation is trying to achieve, but this does affect things a bit. Previously, sorting was done before quantisation. Now, it is done after.

Comment (boks1971) [reply pkg/sfu/audio/audiolevel.go]: Okay, have removed this quantisation as the purpose of that is not clear to me. Can re-introduce if it does serve a real purpose.

**Review by davidzhao — APPROVED** [review]
looks good. it's a good call to convert to linear level sooner.

quantization was put in by @hn8 earlier on. I believe it was to reduce the amount of flickering when sorting by raw audio levels.

**Comment by boks1971** [comment]
Makes sense. Not sorting was happening before quantisation. Can add it back. But we also run into the issue of sort of same quantised value potentially resulting in different order. Will probably leave it out and watch.

**Comment by davidzhao** [comment]
> Makes sense. But sorting was happening before quantisation. Can add it back. But we also run into the issue of sort of same quantised value potentially resulting in different order. Will probably leave it out and watch.

sounds good!

**Pull request merged**